### PR TITLE
:sparkles: Metrics proto proposal

### PR DESCRIPTION
**Pull request by Munklinde96**

### CODE REVIEW

File: proto/rig/api/v1/metrics/metrics.proto [hunk 1]
```diff
@@ -0,0 +1,23 @@
+syntax = "proto3";
+
+package api.v1.metrics;
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
+
+message Topic {
```
Comment (MatiasFrank) [inline proto/rig/api/v1/metrics/metrics.proto:8]: `topic_id` as well?

File: proto/rig/api/v1/metrics/metrics.proto [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+message Metric {
+  google.protobuf.Timestamp timestamp = 1;
+  double value = 2;
+}
+
+message MetricFull {
+  Metric metric = 1;
+  string topic = 2;
+  map<string, string> tags = 3;
```
Comment (MatiasFrank) [inline proto/rig/api/v1/metrics/metrics.proto]: `topic_id`? Then we don't duplicate the tags per metric

File: proto/rig/api/v1/metrics/service.proto [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  // Read metrics for many topics.
+  rpc ReadMetricsMany(ReadMetricsManyRequest)
+      returns (ReadMetricsManyResponse) {}
+
+  // Aggregate metrics for a topic, bucketed by time.
+  rpc BucketedMetrics(BucketedMetricsRequest)
+      returns (BucketedMetricsResponse) {}
+  // Aggregate metrics for many topics, bucketed by time.
+  rpc BucketedMetricsMany(BucketedMetricsManyRequest)
+      returns (BucketedMetricsManyResponse) {}
```
Comment (MatiasFrank) [inline proto/rig/api/v1/metrics/service.proto:27]: Maybe we shouldn't expose topics in the platform API but instead make it project/env/capsule specific. 
I.e.
```
GetMetricsByCapsule(projectID, environmentID, capsuleID string, from ,to time.Time, agg Aggregation) ([]Metric, error)
```
As the public API. Then we do the topic lookup stuff internally. The reason is then we handle lookups with rbac entirely within the platform with some hardcoded logic as to how we map a project,env,capsule triple to topics tags.

Comment (MatiasFrank) [reply proto/rig/api/v1/metrics/service.proto:27]:
```
rpc GetMetrics(GetMetricsRequest) GetMetricsResponse

message GetMetricsRequest {
  string project = 1;
  string env = 2;
  string capsule = 3;
  string metric_type = 4;
  Timestamp from = 5;
  Timestamp to = 6;
  Aggregation aggregation = 7;
  Duration bucket_size = 8;
}

message GetMetricsResponse {
  repeated Metric metrics = 1;
}
```

Comment (MatiasFrank) [reply proto/rig/api/v1/metrics/service.proto:27]: Then we can do validation, e.g. `metric_type` must be given, but `project, env,capsule` are optional (and thus wildcardable)

Comment (Munklinde96) [reply proto/rig/api/v1/metrics/service.proto:27]: yes makes sense

Comment (Munklinde96) [reply proto/rig/api/v1/metrics/service.proto:27]: or wouldn't it make sense, to get an idea of what metrics exist before starting to fetch and process them?

**Pull request merged**